src/recycle/shared_pool.hpp

```diff
@@ -267,6 +285,34 @@ class shared_pool
 
             lock_type lock(m_mutex);
             m_free_list.push_back(resource);
+            m_size--;
+        }
+
+        constexpr std::size_t capacity () noexcept
+        {
+            lock_type lock(m_mutex);
+            return m_max_capacity;
+        }
+
+        constexpr void reserve (std::size_t new_cap)
+        {
+            std::list<value_ptr> pending_resources;
```

Comment:
I think I could use std::deque<> here, since we don't need to any random access.